### DISCUSSION TITLE
Project config-like versionable and non-database reliant feed creation/sharing

### DISCUSSION THREAD

**Discussion by iparr** [engram-design/FeedMe#1350] [Ideas]
_This is a feature request._

The [project config workflow](https://craftcms.com/docs/4.x/project-config.html) in Craft CMS is pretty fun. It makes life easier for developers iterating through live sites and multiple dev environments.

Currently - afaik - configuration of feeds in FeedMe is stored only in the database. Due to the power of this plugin and how it deals with sometimes quite elaborate architectures, the setup can be a bit arduous. The impact of this is felt more keenly when this setup needs to be repeated.

Would be possible to save a feed config (using YAML files like in Project config, rather than just in the DB) in a manner that could be committed to version control instead?

**Comment by tricki** [comment]
I had the same requirement, because the imports are mirrored on multiple deployments of the same site. 

I managed to cobble something together that (mostly) works:

1. Configure the feed(s) through the UI
2. Add code to log the parsed config (see below)
3. Save a feed again, which will log the feed configuration
4. Add the config from Craft's log file (which first needs to be converted from JSON to PHP) to `./config/feed-me.php` (under `feedOptions`), see https://docs.craftcms.com/feed-me/v4/get-started/configuration.html
5. Commit this file and pull the changes on another deployment
6. Create a new feed through the UI
7. On the initial "create" screen, fill out the mandatory fields with any valid data (will be removed immediately after)
8. Save the feed. The values will now be replaced with the values from the config file.
9. Complete the feed creation. All fields should already be filled out correctly.

## Logging the feed config

Add this to `./modules/Module.php::init()`:

```php
Event::on(Feeds::class, Feeds::EVENT_AFTER_SAVE_FEED, function(FeedEvent $event) {
    Craft::info(['FeedEvent After', __METHOD__, json_encode($event->feed)]);
});
```

## Problems with this solution

1. The config might have some things (e.g. IDs) which might not be identical in each environment. I simply removed them from the config file and had to configure them manually.
2. The imports don't seem to work with the `feedOptions` configured. I solved it in two ways:
    1. Initially I simply removed them when all the imports were created everywhere. But when there were new changes that was becoming tedious.
    2. I ended up creating a UI to set a boolean config value which determines whether the `feedOptions` array should be filled. I activate it whenever I have changes in the config which need to go into the database

It still requires some manual work when creating the config but worked well enough, saving me from having to configure dozens of field imports each time.

**Comment by iparr** [comment]
I _think_ this got accommodated to some extent by a paid plugin: https://github.com/litzinger/feed-me-migrations
